app/src/main/java/org/hackillinois/android/viewmodel/ScannerViewModel.kt

```diff
@@ -20,17 +20,29 @@ import retrofit2.Response
 class ScannerViewModel : ViewModel() {
     private val eventRepository = EventRepository.instance
     var eventsListLiveData: LiveData<List<Event>> = MutableLiveData<List<Event>>()
-    var lastScanStatus: MutableLiveData<ScanStatus> = MutableLiveData<ScanStatus>()
+    var lastScanStatus: MutableLiveData<ScanStatus> = MutableLiveData()
     var shouldDisplayOverrideSwitch = MutableLiveData<Boolean>()
 
-    val CHECK_IN_TEXT = "Check In"
+    private val CHECK_IN_TEXT = "Check In"
 
     fun init() {
         eventsListLiveData = eventRepository.fetchAllEvents()
         // Hidden by default
         shouldDisplayOverrideSwitch.postValue(false)
     }
 
+    fun checkUserIntoEvent(eventName: String, userId: String, staffOverride: Boolean) {
```

Comment:
Is it necessary to wrap variables in `CheckIn` and `UserEventPair` objects here? I don't see any benefit to this.

Reply:
Yes, this is the data format accepted by the API. I think retrofit needs the value passed in a single object.

We could construct the objects inside `checkInUser()` and `markUserAsAttendingEvent()` instead, but I don't see much point.